modules/integrations/arrow-integrations-retrofit-adapter/src/main/kotlin/arrow.integrations.retrofit.adapter/IO2CallAdapter.kt

```diff
@@ -0,0 +1,12 @@
+package arrow.integrations.retrofit.adapter
+
+import arrow.effects.IO
+import retrofit2.Call
+import retrofit2.CallAdapter
+import java.lang.reflect.Type
+
+class IO2CallAdapter<R>(private val type: Type) : CallAdapter<R, IO<R>> {
```

Comment:
I believe you can define these adapters polymorphically so they can support a lot more data types than just Call and IO. Something along the lines of:
```kotlin
class GenericCallAdapter<F, A>(private val type: Type, private val delay: Async<F>) : CallAdapter<A, Kind<F, A>>, Async<F> by delay {
  override fun adapt(call: Call<A>): Kind<F, A> = async { proc -> call.enqueue(ResponseCallback(proc)) }
   override fun responseType(): Type = type
}
```

Reply:
You may then make them concrete where needed by passing in the async instance of each of the data types you wish to support. With a generic adapter you may already be supporting Observable, Flux, Deferred and many others

Reply:
That's what the other adapter, `CallK`, does. We cannot create and register an adapter because retrofit is implemented by doing runtime checks on a Class return using reflection, and it doesn't get all the generics we need for us to create our own internally. `CallK` fixes that, check the code.

Reply:
If `CallK` is already polymorphic why do we need then an `IO` call adapter? Couldn;t the user use the `CallK` adapter to get back responses on `IO`?

Reply:
To avoid the extra step if users want to. I'm on the fence about it because we have to decide which semantics we'll need to enforce for it.

Reply:
The way I see it this needs `arrow-retrofit-coroutines`, `arrow-retrofit-rx2`, etc. for each one of the currently supported effect types and with those modules you get the specialization for the runtime you are gonna support

Reply:
My idea for the IO adapter is to create a way for a developer be able to use the plugin without understanding how to create polymorphic code, which is not a very simple concept. 
I don't mind removing the adapter, but we should keep in mind that documentation must teach how to concretise the type for someone that is not very used advanced concepts.